src/downloader/gh_api.py

```diff
@@ -190,56 +196,108 @@ def get_repository_workflows(repo: str) -> Dict[str, str]:
+def get_commit_sha_of_ref(repo: str, ref: str) -> Optional[str]:
+    """
+    This function is used to get the commit sha of a ref or tag in a repository.
+    It will first check if it got ref, if not, it will return the commit sha of the latest commit.
+    If it got a ref, it will check if it is a tag, and if so, it will return the commit sha of the tag.
+    Then, if the ref is a branch name (and not a tag), it will return the latest commit sha in the branch.
+    Otherwise it will return ref as is which is a commit sha.
+
+    If got error in the process, it will return None
+    """
+    if ref is None:
+        # If no ref is provided, we return the latest commit sha
+        r = get(COMMITS_URL.format(repo_path=repo), headers=headers)
+        if r.status_code != 200:
+            log.error(
+                f"Coudln't found commits for repository {repo}. status code: {r.status_code}. Response: {r.text}"
+            )
+            return
+
+        return r.json()[0]["sha"]
+
+    # Check if it is a tag
+    r = get(TAGS_URL.format(repo_path=repo), headers=headers)
+    if r.status_code == 200:
+        shas = [d["commit"]["sha"] for d in r.json() if d["name"] == ref]
 
-    receives 'path_in_repo' relative path to the repository root to where search the action.yml.
-    It should be a directory and not a file. (if file this is a reusable workflow)
+        # Sometimes the tag is not found, but the provided reference is actually a branch name.
+        if len(shas) > 0:
+            return shas[0]
 
-    Raises exception if network error occured.
+    # Check if it is a branch
+    r = get(BRANCHES_URL.format(repo_path=repo, branch=ref), headers=headers)
+    if r.status_code == 200:
+        return r.json()["commit"]["sha"]
+
+    # If we got here, it means that ref is a commit sha
+    return ref
+
+
+def get_download_url(
+    path: str,
+    ref: Union[str, None],
+    file_suffixes: Union[List[str], None],
+) -> Optional[Tuple[str, str]]:
+    """
+    Retrieves the downloadable URL for a GitHub resource located at the given path.
+
+    Parameters:
+    - path (str): The repository path containing the resource, formatted as "owner/repo/relative_path_to_resource".
+    - ref (Optional[str]): The ref or the tag of the resource. If None, the latest version is used.
+    - file_suffixes (List[str]): List of possible file suffixes that the resource could have (e.g., ["action.yml", "action.yaml"]).
+
+    Returns:
+    Tuple of the downloadable URL for the resource and the commit sha. Returns None if the resource is not found or if a network error occurs.
+
+    Raises:
+    - Logs an error message if the request to the GitHub API fails.
     """
-    path_splitted = path.split("/")
-    repo = "/".join(path_splitted[:2])
-    relative_path = "/".join(path_splitted[2:])
+    splitted_path = path.split("/")
+    repo, relative_path = "/".join(splitted_path[:2]), "/".join(splitted_path[2:])
```

Comment:
Maybe this could be a util function that will get a little bit of documentation?